### ISSUE TITLE
Crosh shell not opening - illegal Instruction (core dumped)

### ISSUE THREAD

**Issue by joeyoropesa-dev** [sebanc/brunch#1362]
When I try to open Crosh shell with keyboard, crosh shell constantly close itself. After that when I try run crosh shell from terminal (Ctrl + Alt + F2) I get this error:

illegal instruction (core dumped)

I have latest Brunch and ChromeOS version so - is this bug by Brunch or by Chrome OS. And if this bug by Brunch framework, can this problem be solved. My Chrome OS image is rammus type. I don't have installed anything on my device (Android, Crostini - nothing)

/var/log/messages says:
2022-01-02T10:10:00.694175Z INFO kernel: [ 2515.077547] traps: crosh[9981] trap invalid opcode ip:59d35b93c4d9 sp:7ffeb42c8600 error:0 in crosh[59d35b844000+1e3000]
2022-01-02T10:10:01.196282Z WARNING crash_reporter[9984]: [user] Received crash notification for crosh[9981] sig 4, user 0 group 0 (handling)
2022-01-02T10:10:01.197404Z INFO crash_reporter[9984]: State of crashed process [9981]: S (sleeping)
2022-01-02T10:10:01.198684Z INFO crash_reporter[9984]: Accessing crash dir '/var/spool/crash' via symlinked handle '/proc/self/fd/11'
2022-01-02T10:10:01.219384Z WARNING crash_reporter[9984]: exec name 'crosh' not found in log file
2022-01-02T10:10:01.303526Z INFO crash_reporter[9984]: Stored minidump to /var/spool/crash/crosh.20220102.111001.28129.9981.dmp
2022-01-02T10:10:01.303882Z INFO crash_reporter[9984]: Finishing crash. Meta file: /proc/self/fd/11/crosh.20220102.111001.28129.9981.meta

**Comment by tejasraman** [comment]
Best I can think of is that Brunch is sending illegal(unsupported) instructions to your CPU. What exactly is your CPU model? Maybe try a different recovery image, "caroline" or "syndra" might work?

**Comment by tejasraman** [comment]
Best I can think of is that Brunch is sending illegal(unsupported) instructions to your CPU. What exactly is your CPU model? Maybe try a different recovery image, "syndra" or "syndra" might work? - FYI I'm not sebanc, and it could be something else

**Comment by BCDTechPals** [comment]
I meant "syndra" or "caroline".

**Comment by tejasraman** [comment]
I meant "syndra" or "caroline" - just to try :)

**Comment by admin-elvistrujanovic** [comment]
> Best I can think of is that Brunch is sending illegal(unsupported) instructions to your CPU. What exactly is your CPU model? Maybe try a different recovery image, "syndra" or "syndra" might work? - FYI I'm not sebanc, and it could be something else

Processor: Intel(R) Celeron(R) CPU B820 @ 1.70GHz

**Comment by tejasraman** [comment]
That's Sandy Bridge. There shouldn't be problems with Sandy Bridge, but a "fairly recent" Celeron is recommended. You could try "nami" recovery image, or use "caroline" (try caroline first). It may just be that the _rammus_ image isusing unsupported instruction sets for 2013< - there is supposedly a workaround inside Brunch, may not always work

Here is the caroline image: [https://cros.tech/device/caroline](https://cros.tech/device/caroline)
and the nami one: [https://cros.tech/device/nami](https://cros.tech/device/nami)

**Comment by tejasraman** [comment]
That's Sandy Bridge. There shouldn't be problems with Sandy Bridge, but a "fairly recent" Celeron is recommended. You could try "nami" recovery image, or use "caroline" (try caroline first). It may just be that old issue that the _rammus_ image is using unsupported instruction sets for 2013< - there is supposedly a workaround inside Brunch, may not always work

Here is the caroline image: [https://cros.tech/device/caroline](https://cros.tech/device/caroline)
and the nami one: [https://cros.tech/device/nami](https://cros.tech/device/nami)

**Comment by tejasraman** [comment]
That's Sandy Bridge. There shouldn't be problems with Sandy Bridge, but a "fairly recent" Celeron is recommended. You could try "nami" recovery image, or use "caroline" (try caroline first). It may just be that the _rammus_ image is using unsupported instruction sets. Try these two and get back to me if they don;'t work :)
Here is the caroline image: [https://cros.tech/device/caroline](https://cros.tech/device/caroline)
and the nami one: [https://cros.tech/device/nami](https://cros.tech/device/nami)

**Comment by tejasraman** [comment]
Sorry for comment-bombing, but you may have to use "samus" (which doesn't get updates anymore) which is about the same era as your Celeron if name or Caroline don't work

**Comment by tejasraman** [comment]
Sorry for comment-bombing, but you might have to use "samus" (which doesn't get updates anymore) which is about the same era as your Celeron if older images like nami or caroline don't work

**Comment by tejasraman** [comment]
Sorry for comment-bombing, but you might have to use "samus" or "lulu" (which don't get updates anymore). They are about 3 yrs. newer than your Celeron if newer images like nami or caroline don't work

**Comment by tejasraman** [comment]
Sorry for comment-bombing, but you might have to use "samus" or "lulu" (which don't get updates anymore) if caroline or nami don't work. They are about 3 yrs. newer than your Celeron.

**Comment by tejasraman** [comment]
FYI, the Brunch page says that Baytrail or later is needed - you have the older "Sandy Bridge". You could try a different recovery ("nami" or "caroline") and if they work, you could try "[samus](https://cros-updates-serving.appspot.com//#samus)" (if you're OK with a release of Chrome OS from June 2021) because that supported your CPU.

**Comment by tejasraman** [comment]
FYI, the Brunch page says that Baytrail or later is needed - you have the older "Sandy Bridge". You could try a different recovery ("nami" or "caroline") and if they don't work, you could try "samus" (if you're OK with a release of Chrome OS from June 2021) because that supported your CPU.

**Comment by tejasraman** [comment]
FYI, the Brunch page says that Baytrail or later is needed - you have the older "Sandy Bridge". You could try a different recovery ("nami" or "caroline") and if they work, you could try "samus" (if you're OK with a release of Chrome OS from June 2021) - that has a higher chance, but is now end of life.

**Comment by admin-elvistrujanovic** [comment]
> FYI, the Brunch page says that Baytrail or later is needed - you have the older "Sandy Bridge". You could try a different recovery ("nami" or "caroline") and if they work, you could try "samus" (if you're OK with a release of Chrome OS from June 2021) - that has a higher chance, but is now end of life.
> 
> 
Is "nami" image supported by Google and my CPU?

**Comment by tejasraman** [comment]
Yes, the “nami” image is supported until 2025 (rammus hits EOL in 2026 - a year after nami) but nami should have better support for your CPU. This is an honestly weird error, and usually, these weird errors on seemingly supported CPUs (this is the actual Sandy Bridge architecture) can be solved using a different recovery.

**Comment by tejasraman** [comment]
FYI, the Brunch page says that Baytrail or later is needed - you have the older "Sandy Bridge". You could try a different recovery ("nami" or "caroline") and if they **don’t** work, you could try "samus" (if you're OK with a release of Chrome OS from June 2021) - that has a higher chance, but is now end of life - _sorry for the confusion with this comment_

**Comment by admin-elvistrujanovic** [comment]
> Yes, the “nami” image is supported until 2025 (rammus hits EOL in 2026 - a year after nami) but nami should have better support for your CPU. This is an honestly weird error, and usually, these weird errors on seemingly supported CPUs (this is the actual Sandy Bridge architecture) can be solved using a different recovery.

On stable and beta version, crosh problem is resolved but in dev - the problem is the same. Maybe, if I change the kernel in the brunch settings for booting this image - i can use crosh on the dev version too.

**Comment by tejasraman** [comment]
Try kernel 4.19 - your CPU is more likely to have better support

**Comment by tejasraman** [comment]
Ohhhh, you’re on dev builds. Do you have the Brunch PWA installed?

**Comment by tejasraman** [comment]
fyi I am asking just because for dev, you need brunch-unstable installed

**Comment by tejasraman** [comment]
Ohhhh… you are on ChromeOS Dev builds. You usually need brunch-unstable to use dev builds (it contains the latest Brunch code). Do you have the Brunch PWA installed on your Brunch laptop?

**Comment by tejasraman** [comment]
That’s how you change from brunch-stable to brunch-unstable(brunch dev releases)

**Comment by admin-elvistrujanovic** [comment]
> Ohhhh… you are on ChromeOS Dev builds. You usually need brunch-unstable to use dev builds (it contains the latest Brunch code). Do you have the Brunch PWA installed on your Brunch laptop?

I'm using Brunch Unstable

**Comment by tejasraman** [comment]
Maybe a messed up grub config. Press E on the Brunch option and see what is going on. Tell me what shows up in the grub editor.

**Comment by tejasraman** [comment]
FYI. I still think trying nami would fix it. Yours sounds like unrecognized hardware/CPU errors.

**Comment by admin-elvistrujanovic** [comment]
> Maybe a messed up grub config. Press E on the Brunch option and see what is going on. Tell me what shows up in the grub editor.

Fixed. When I changed brunch kernel from 5.15 to 5.10, it's working now normally

**Comment by tejasraman** [comment]
Great thanks!

**Comment by ganeshambalakkara** [comment]
I'm too having the same problem crosh closes as soon as i press Ctrl+Alt+T.
After that when I try run crosh shell from terminal (Ctrl + Alt + F2) I get this error:
 illegal instruction (core dumped)

**Comment by ganeshambalakkara** [comment]
@admin-elvistrujanovic Can please help me how u solved this issue

**Comment by tejasraman** [comment]
Don't use kernel 5.15

**Comment by ganeshambalakkara** [comment]
> 

I'm using kernel 5.4 only still

**Comment by ganeshambalakkara** [comment]
@admin-elvistrujanovic I'm using kernel 5.4 only still

**Comment by tejasraman** [comment]
What is your computer?

**Comment by ganeshambalakkara** [comment]
> What is your computer?
> […](#)
> On May 23, 2022, 1:03 PM -0600, ganeshambalakkara ***@***.***>, wrote: @admin-elvistrujanovic I'm using kernel 5.4 only still — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you commented.Message ID: ***@***.***>
Mine is hp elitebook 8460p 
320GB hardisk 4gb DDR2 ram with integrated graphics intel i5

**Comment by tejasraman** [comment]
DDR2?
This laptop should work with Rammus, and it seems that your hardware is similar to of @admin-elvistrujanovic.

FYI I am NOT @admin-elvistrujanovic.

**Comment by tejasraman** [comment]
Warning: Don't update to r101 on your model
Try kernel 5.10

**Comment by ganeshambalakkara** [comment]
@tejasraman would i be able to update to another version?

**Comment by tejasraman** [comment]
Not until @sebanc fixes the Mesa driver issue.
Try kernel 5.10 in Brunch config, **do not upgrade to r101**

**Comment by soupCodez** [comment]
I have this same issue

**Comment by tejasraman** [comment]
Hello @soupCodez !
We need some more context to deal with this. Could you please tell me
- Your chrome OS version
- Your Brunch version
- Your CPU, GPU, and computer model (if it is a laptop/prebuilt PC)
- Your kernel and any other configuration options set up

(Ya, that was a bit generic, sorry - but I can't really do much until I know info)

**Comment by tejasraman** [comment]
@soupCodez are you still having this issue?

**Comment by dhomas1** [comment]
Hello there!

I am still having this problem. I am on Brunch r103 with the image for Rammus on ChromeOS v103. My hardware is an HP Probook 67470b that uses a Core i5 3rd gen processor. I am using Kernel 5.10, but I had the same behaviour with kernel 5.4 and 5.14. 

I have Brunch installed on my USB key using the instructions from Github (not any other video instructions). When I get to the part where I want to install it to the HDD, I cannot open crosh. 

Troubleshooting steps I've tried:
- different kernel versions
-  disabling nX in the BIOS
- trying different boot options in Brunch Configuration Menu
- tried installing to the HDD via sudo from the root shell (CTRL - ALT - F2), which allowed me to install, but I still had crosh crashing on start. Android apps also can be installed but don't work, which I assume is a side effect of crosh not working

Any help would be welcome. Thank you!

**Comment by tejasraman** [comment]
Hi @dhomas1 
Seems you're very thorough in your troubleshooting (great job!) but there is something wrong (it seems).

1. Are you using bios or EFI?
2. Have you reinstalled CrOS even once?
> installing Chrome OS v104 with Brunch r103 (I didn't think this would work :) )
It works most of the time 😉
3. Android is unrelated - try Android init fix or enabling VT-D (virtualization) in BIOS setup

**Comment by dhomas1** [comment]
@tejasraman Thanks for the tips!

0: I've been using a ChromeOS USB key, using the instructions here: https://github.com/sebanc/brunch/blob/master/install-with-windows.md

I'll try using Ubuntu instead.

1: I've set up the image without the MBR patch. My Boot mode is set to "UEFI Hybrid (with CSM)", though I can set it up to "UEFI Native (without CSM)" to see if it helps.

2: I've installed and reinstalled CrOS multiple times, though I've only used the Rammus image (not sure which other image would be compatible with my hardware). 

3: I've set up the Android init fix, but my apps all still seem to crash.

I said I'd disabled nX (No eXecute), but in reality it was "Data Execution Prevention" that I'd disabled. In any case, it didn't change the outcome.

I'll try the changes and report back. Thanks for the advice!

**Comment by tejasraman** [comment]
Use it without the BIOS patch for better boot times and less problems. The BIOS patch IMO is nothing but trouble. CSM is ok but don't use the patch.

You can use balenaEtcher to burn Ubuntu onto a USB

**Comment by tejasraman** [comment]
Hi @dhomas1 ! I can provide you with an (up to date) video tutorial if you want.

Tejas

**Comment by tejasraman** [comment]
@dhomas1 have you managed to get it working?

**Comment by Not-Dhanraj** [comment]
hey there @tejasraman . Can you suggest brunch type for Intel(R) Celeron(R) CPU B830 ?
Thanks.
It would be amazing if you could also include a tutorial.

**Comment by tejasraman** [comment]
The best I can give you is Caroline.

I'll give you a tutorial but I need to know if you're using windows or Linux.

**Comment by tejasraman** [comment]
If you see #1677 I'm not sure if your celeron will work however it might. If you could try [Caroline](https://cros.tech/device/caroline) + [brunch 107-unstable](url)https://github.com/sebanc/brunch-unstable/releases/tag/r107-unstable-20221029) that might work.

You can read the official docs. If you want singleboot I can help but dualboot is a mixed bag (especially when it comes to fedora)

**Comment by Not-Dhanraj** [comment]
> 

I don't intend to ise btw line anyways lamo😂
 I use arch cause it is pretty lightweight on my b830 laptop. Tried debian as well but arch felt smooth so going with it.
Haven't updated/turned it on from months now i need to chroot and update everything 😂

**Comment by Not-Dhanraj** [comment]
> If you see #1677 I'm not sure if your celeron will work however it might. If you could try [Caroline](https://cros.tech/device/caroline) + [brunch 107-unstable](url)https://github.com/sebanc/brunch-unstable/releases/tag/r107-unstable-20221029) that might work.
> 
> You can read the official docs. If you want singleboot I can help but dualboot is a mixed bag (especially when it comes to fedora)

Yeah, i want singleboot. If it'll work i'll upgrade to a 150 gig ssd.
Are you sure that unstable might not be unstable? 🤔
Can it work with stable?

**Comment by tejasraman** [comment]
I mean it will but unstable is never unstable. I wanted to replicate my Sandy bridge setup but stable _should_ work.

Single boot :)

(BTW I **don't** use Arch)

**Comment by tejasraman** [comment]
I mean it will but unstable is never unstable. I wanted to replicate my Sandy bridge setup but stable _should_ work.

Single boot :) I can help 

(BTW I **don't** use Arch)

**Comment by tejasraman** [comment]
I'll give you a tutorial in any form you want.

**Comment by Not-Dhanraj** [comment]
> I'll give you a tutorial in any form you want.

Video tutorial would be amazing 😅

**Comment by tejasraman** [comment]
Will do! Want me to do it on arch?

**Comment by Not-Dhanraj** [comment]
> Will do! Want me to do it on arch?

Do it with whatever you have right now, or whatever you are comfortable with😁

**Comment by tejasraman** [comment]
I have an arch partition on my main PC

**Comment by Not-Dhanraj** [comment]
Thanks!

**Comment by tejasraman** [comment]
I'm still trying to make this into a video tutorial but here is a text tutorial:
- Install [Chaotic](https://aur.chaotic.cx) Arch repos
- Run `sudo pacman -S pv vboot-utils e2fsprogs`
- Mount the drive, format it to ext4, and run `tune2fs -m 0 /dev/sdX` (for a PC of that generation it's probably sata) **(Only needed for very small disks)**
- Download the image "Caroline" from [cros.tech](https://cros.tech/device/caroline)
- Download brunch unstable 107
- Extract both zips into a folder
- Run `sudo bash chromeos-install.sh -src your image.bin -dst mountedlocation`

PS the computer needs EFI support for this to work. I'll see if the legacy bios patch still works on newer brunch versions.

**Comment by Not-Dhanraj** [comment]
Dont worry abour legacy bios, the laptop in which i am trying to install chrome os supports EFI

**Comment by tejasraman** [comment]
Great... The only Sandy bridge laptop I have is a MacBook with EFI so...

Did that work for you?

**Comment by jmbocn** [comment]
Hi,

same issue here, I'm running nami r100 over brunch r107 unstable 20221029 on a core i3 3217U processor. Tried different kernels, now on 5.4, tried also ramus r100, crosh keeps on crashing with illegal instruction.

Any suggestions welcome!

**Comment by tejasraman** [comment]
Could you try Caroline @jmbocn

**Comment by jmbocn** [comment]
Hi Tejas, same issue (tried both r100 and r96); plus I cannot login to the system on the graphical GUI. If I switch to terminal mode with CTRL+ALT+F2 I can (1) check that crosh crashes with illegal instruction and (2) on /var/log/messages I see recurrent crashes with lots of ERR from missived calling method .....UploadEncryptedRecord (possibly related to the unability to log in). I reverted to nami r100 which is functional for me -other than crosh-.

**Comment by jmbocn** [comment]
Update: I realized I **_can_** run crosh, just not interactively. E. g.: `crosh -- <command>` works but just `crosh` will fail. I guess that's why CTRL + ALT + F2 just opens and quickly closes the window, because crosh crashes if there are no arguments, that is if I try to execute crosh interactively to get a `crosh>` prompt

**Comment by Not-Dhanraj** [comment]
> Great... The only Sandy bridge laptop I have is a MacBook with EFI so I never tested the patch.
> 
> Did the guide work for you?
> 
> _Edit: The reason I mentioned it was that most Windows Sandy Bridge laptops had Windows 7 (no EFI)._

Yes it did but sadly internet connection wasn't working so installed mint

**Comment by tejasraman** [comment]
What's your laptop model? I'll try to help

**Comment by Not-Dhanraj** [comment]
It is hp notebook 2000. And i cannot tell exact model number right now as i don't have access to that device right now.

**Comment by Not-Dhanraj** [comment]
When installed chromeos then it connected to wifi just fine and my network was working fine as well but it wasnt opening any website + i was not able to login with my google account.

Linux development tools wasn't supported on that device either, so i lost interest as soon as i installed it😂

**Comment by tejasraman** [comment]
You know I hate chromeOS. My school forces a piece of junk Chromebook on me and it's limited (crostini = slow). 

At the same time brunch is an interesting project (for me) to mess around with. (I even added Linux 6.0 to it recently).

I understand what you mean but thanks so much for being so helpful!

**Comment by tejasraman** [comment]
That's weird especially because on an i5-2435M crosh works fine. Your chip has the same/more instruction sets than mine. @admin-elvistrujanovic had this but sadly IvyBridge is not supported (works though).

Can you test r107 unstable?

**Comment by tejasraman** [comment]
Then I'm not sure anymore.
